### PR TITLE
Add a `pending_delete` field to projects and groups

### PR DESCRIPTION
**Pull request by svanharmelen**

### CODE REVIEW

**Comment by svanharmelen** [comment]
I think this adds the missing bits in order to be able to better support soft deleted project in the TF provider.

I'm not in a position to test this, so please let me know if the fields are properly populated so they can be used to determine if a project create failed because it already exists, or because it is currently soft deleted.

Fixes #782

File: projects.go [hunk 1]
```diff
@@ -94,9 +95,10 @@ type Project struct {
 	} `json:"shared_with_groups"`
 	Statistics        *ProjectStatistics `json:"statistics"`
 	Links             *Links             `json:"_links,omitempty"`
-	CIConfigPath      *string            `json:"ci_config_path"`
+	CIConfigPath      string             `json:"ci_config_path"`
```
Comment (ringods) [inline projects.go]: @svanharmelen Is this an intentional change as I don't think it is related to the "pending delete"

Comment (svanharmelen) [reply projects.go]: It’s not related but it is intentional. Could make a dedicated PR for it, but thought to just add it when I noticed it...

Comment (ringods) [reply projects.go]: You don't have to put this in a dedicated PR. I just want to rule out any accidental mistake. ;-)

Comment (svanharmelen) [reply projects.go]: Thanks for bringing it up 👍🏻

**Pull request merged**